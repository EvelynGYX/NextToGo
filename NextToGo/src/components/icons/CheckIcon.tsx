import React from 'react';
import {StyleProp, ViewStyle} from 'react-native';
import FontAwesome5Icon from 'react-native-vector-icons/FontAwesome5';

export interface CheckIconProps {
  isChecked: boolean;
  onPress: () => void;
  style?: StyleProp<ViewStyle>;
}

const CheckIcon = (props: CheckIconProps) => {
  const isChecked = (isSelected: boolean): string => {
    if (isSelected) {
      return 'check-circle';
    } else {
      return 'circle';
    }
  };

  return (
    <FontAwesome5Icon
      style={props.style}
      name={isChecked(props.isChecked)}
      onPress={props.onPress}
      size={20}
    />
  );
};

export default React.memo(CheckIcon);
