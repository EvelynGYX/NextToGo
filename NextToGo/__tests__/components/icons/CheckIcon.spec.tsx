import {shallow} from 'enzyme';
import React from 'react';
import {StyleProp, ViewStyle} from 'react-native';
import FontAwesome5Icon from 'react-native-vector-icons/FontAwesome5';
import renderer from 'react-test-renderer';
import CheckIcon, {
  CheckIconProps,
} from '../../../src/components/icons/CheckIcon';

describe('CheckIcon', () => {
  const mockStyle: StyleProp<ViewStyle> = {height: 100};

  let mockProps: CheckIconProps = {
    isChecked: 'some isChecked' as any as boolean,
    onPress: jest.fn(),
    style: mockStyle,
  };

  it('renders correctly', () => {
    const tree = renderer.create(<CheckIcon {...mockProps} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders check-circle if isSelected prop is true', () => {
    mockProps.isChecked = true;
    const wrapper = shallow(<CheckIcon {...mockProps} />);
    expect(wrapper.find(FontAwesome5Icon).props().name).toEqual('check-circle');
  });

  it('renders circle if isSelected prop is false', () => {
    mockProps.isChecked = true;
    const wrapper = shallow(<CheckIcon {...mockProps} />);
    expect(wrapper.find(FontAwesome5Icon).props().name).toEqual('check-circle');
  });
});
