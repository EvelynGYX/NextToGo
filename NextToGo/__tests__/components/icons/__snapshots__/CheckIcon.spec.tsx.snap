// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckIcon renders correctly 1`] = `
<Text
  allowFontScaling={false}
  onPress={[MockFunction]}
  selectable={false}
  style={
    Array [
      Object {
        "color": undefined,
        "fontSize": 20,
      },
      Object {
        "height": 100,
      },
      Object {
        "fontFamily": "FontAwesome5Free-Regular",
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
      Object {
        "fontWeight": "400",
      },
    ]
  }
>
  
</Text>
`;
